search/includes/classes/class-versioning.php

```diff
@@ -15,6 +15,40 @@ class Versioning {
 	 */
 	private $current_index_version_by_type = array();
 
+	/**
+	 * An internal record of every object that has been queued up (via Queue::queue_object()) so that we can replicate those jobs
+	 * to the other index versions at the end of the request
+	 */
+	private $queued_objects_by_type_and_version = array();
+
+	/**
+	 * Internal flag to prevent infinite loops when handling object deletions
+	 */
+	private $is_doing_object_delete;
+
+	public function __construct() {
+		// When objects are added to the queue, we want to replicate that out to all index versions, to keep them in sync
+		add_action( 'vip_search_indexing_object_queued', [ $this, 'action__vip_search_indexing_object_queued' ], 10, 4 );
+		add_action( 'shutdown', [ $this, 'action__shutdown' ] );
```

Comment:
That's a good idea - also we need to be sure we always run _after_ EP's shutdown hook, otherwise the `pre_ep_index_sync_queue` filter may not have fired and we won't have found any posts to replicate. 👍

Reply:
Moved the versioning shutdown hook to `100` in e3c7e2e01